comp/core/flare/component.go

```diff
@@ -52,11 +50,13 @@ type Mock interface {
 	GetFlareFile(t *testing.T, filename string) (string, error)
 }
 
-// re-exports (avoiding a Go package dependency loop)
-
-type Registration = reg.Registration
+// Registration is provided by other components to register themselves to
+// provide flare data.
+type Registration struct {
+	fx.Out
 
-var FileRegistration = reg.FileRegistration
+	Registration registration `group:"true"`
```

Comment:
I made that "true" when the group registration logic wasn't hidden, because it was one less thing to get right.  But you're correct -- it could be anything now, as long as it agrees.  I can update the docs.

Reply:
d5b48ed4beca3ce3c2f2bd4260bebe4ec73ce93b